### PR TITLE
feat!: export crypto primitives

### PR DESCRIPTION
**Pull request by weboko**
## Problem

In some cases users don't want to encrypt whole `payload` but only some part of it to have additional features.
To do some one would need to implement encryption on their own or find a library.

## Solution

Export already implemented primitives to allows consumers build more features.

## Notes

- Related to https://github.com/waku-org/js-waku-examples/pull/286

## Follow-up

- [ ] update https://github.com/waku-org/js-waku-examples/pull/286

### CODE REVIEW

**Comment by weboko** [comment]
This PR is mostly re-naming and exporting of existing functions.

File: packages/message-encryption/src/misc.ts [hunk 1]
```diff
@@ -8,3 +8,12 @@ export const Symmetric = {
...
 };
+
+export const OneMillion = BigInt(1_000_000);
+
+export const Version = 1;
+
+export type Signature = {
+  signature: Uint8Array;
+  publicKey: Uint8Array | undefined;
+};
```
[the remark this answers is not in the document]
Comment (weboko) [reply packages/message-encryption/src/misc.ts]: good spot, moved

File: packages/message-encryption/src/crypto/crypto.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
+import nodeCrypto from "crypto";
```
[the remark this answers is not in the document]
Comment (weboko) [reply packages/message-encryption/src/crypto/crypto.ts]: indeed looks strange, renamed to utils, can't think of anything better

**Comment by fryorcraken** [comment]
@waku-org/js-waku-developers I believe this PR should be reverted as it encourages metadata leaking by enabling developers to partial message encryption.

**Pull request merged**